### PR TITLE
DrivAerML: moderate WD=5e-3 + gc=1.0 at T_max=30 — gentler regularization

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

Your previous compound experiment (#2908) showed that gc=1.0+WD=1e-2 combined with aggressive cosine cycling crashes. The crashes happen at cosine LR peaks where gradients spike. One likely culprit: WD=1e-2 (from AirfRANS golden) may be too aggressive for DrivAerML's larger 4L/512d model — more parameters means the L2 penalty has outsized influence on the gradient magnitude.

This experiment tests **moderate WD=5e-3** — half the AirfRANS value — with gc=1.0, keeping the baseline T_max=30 scheduler (which is stable). We also test WD=5e-3 alone (no gc) to isolate its effect. brook #2878 tests gc=1.0+WD=1e-2, shinobu #2912 tests WD=3e-2/5e-2 — this fills the gap at WD=5e-3.

Additionally: a pure gc=1.0 ablation at baseline config (no WD, no T_max change) to establish the isolated effect of gradient clipping on DrivAerML.

## Instructions

**IMPORTANT: Use EXACTLY these run specifications. All runs use AdamW.**

Run 1 — DrivAerML WD=5e-3 + gc=1.0 at T_max=30 (moderate compound):
```bash
cd target/icml2026 && python train.py --dataset drivaerml --optimizer adamw \
  --lr 5e-4 --cosine-t-max 30 --no-use-ema --enable-fourier \
  --model-layers 4 --model-hidden-dim 512 --model-heads 8 \
  --grad-clip 1.0 --weight-decay 5e-3 \
  --epochs 999 --batch-size 1 \
  --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 --max-eval-batches 200 \
  --wandb-group yuji-moderate-wd
```

Run 2 — DrivAerML WD=5e-3 only (no gc, isolate WD effect):
```bash
cd target/icml2026 && python train.py --dataset drivaerml --optimizer adamw \
  --lr 5e-4 --cosine-t-max 30 --no-use-ema --enable-fourier \
  --model-layers 4 --model-hidden-dim 512 --model-heads 8 \
  --weight-decay 5e-3 \
  --epochs 999 --batch-size 1 \
  --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 --max-eval-batches 200 \
  --wandb-group yuji-moderate-wd
```

Run 3 — DrivAerML gc=1.0 only (no WD, pure gc ablation at baseline config):
```bash
cd target/icml2026 && python train.py --dataset drivaerml --optimizer adamw \
  --lr 5e-4 --cosine-t-max 30 --no-use-ema --enable-fourier \
  --model-layers 4 --model-hidden-dim 512 --model-heads 8 \
  --grad-clip 1.0 \
  --epochs 999 --batch-size 1 \
  --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 --max-eval-batches 200 \
  --wandb-group yuji-moderate-wd
```

Run 4 — AirfRANS WD=5e-3 + gc=1.0 at T_max=5 (cross-dataset: does lower WD help AF too?):
```bash
cd target/icml2026 && python train.py --dataset airfrans --airfrans-task full \
  --optimizer adamw --lr 7e-4 --cosine-t-max 5 \
  --grad-clip 1.0 --weight-decay 5e-3 \
  --no-use-ema --enable-fourier \
  --model-layers 2 --model-hidden-dim 256 --model-heads 4 --epochs 999 \
  --wandb-group yuji-moderate-wd
```

Report `val_primary/surface_rel_l2_pct` for DrivAerML and `val_primary/surface_mse` for AirfRANS. For Run 3, also note whether gc=1.0 alone (without WD) is stable at baseline T_max=30 — this establishes whether gc is inherently helpful for DrivAerML.

## Baseline

- **DrivAerML val_primary/surface_rel_l2_pct:** 4.619% (PR #2691, no gc, no WD, T_max=30)
- **AirfRANS val_primary/surface_mse:** 0.001236 (PR #2828, gc=1.0, WD=1e-2, T_max=5)
- **DrivAerML external target:** 3.71% (AB-UPT)
- **Reproduce golden DrivAerML:** `cd target/icml2026 && python train.py --dataset drivaerml --optimizer adamw --lr 5e-4 --cosine-t-max 30 --no-use-ema --enable-fourier --model-layers 4 --model-hidden-dim 512 --model-heads 8 --epochs 999 --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200`

### CODE REVIEW

**Comment by morganmcg1** [comment]
Closing to narrow the queue for the final ICML benchmark sprint. Current priority is benchmark-specific optimization and benchmark-faithful reporting for DrivAerML, TandemFoil paper benchmark, AirfRANS full Surf/Vol, and the Tandem parity anchor. If this idea still matters later, we can reopen it in a benchmark-specific form.
